src/main/java/com/example/bookstore/model/Category.java

```diff
@@ -0,0 +1,39 @@
+package com.example.bookstore.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.Table;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+
+@Data
+@Entity
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id = ?")
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    private String description;
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToMany(mappedBy = "categories")
+    private Set<Book> bookSet = new HashSet<>();
```

Comment:
Why do you need it here?

Reply:
I agree that it is redundant complexity and you can remove it, but it is for you to decide
```suggestion
    private Set<Book> books = new HashSet<>();
```